scm/github.go

```diff
@@ -689,6 +689,59 @@ func (s *GithubSCM) GetUserScopes(ctx context.Context) *Authorization {
 	return &Authorization{Scopes: gitScopes}
 }
 
+// GetRepositoryInvites implements the SCM interface
+func (s *GithubSCM) GetRepositoryInvites(ctx context.Context, opt *RepositoryInvitationOptions) ([]*RepositoryInvitation, error) {
+	if !opt.valid() {
+		return nil, ErrMissingFields{
+			Method:  "GetRepositoryInvites",
+			Message: fmt.Sprintf("%+v", opt),
+		}
+	}
+
+	invites, _, err := s.client.Users.ListInvitations(ctx, &github.ListOptions{})
```

Comment:
One potential issue with this implementation... what if a student (user) has unrelated invitations that he doesn't want to accept. Will the code below simply accept all invitations, including those unrelated to the user's QuickFeed courses?? We should test this somehow, to understand the behavior of what is returned from `ListInvitations`.

Reply:
From my testing I found that we could filter out unrelated invitations by looking at the owner of the repository invite.

```go
	for _, invite := range invites {
		if invite.Repo.Owner.GetLogin() != opt.Owner || invite.Invitee.GetLogin() != opt.Login {
			// Ignore unrelated invites
			continue
		}
                ...
        }
```

Where the repository in `invite.Repo` looks like this:

```go
// invite.Repo
github.Repository{
    ...
    Owner: github.User{
        Login:"quickfeed-test-org", 
        ...
        HTMLURL:"https://github.com/quickfeed-test-org"
        Type:"Organization"
        ...
    }
    ...
}
```

Example of an unrelated invite:

```go
// Unrelated invite to a repo created by an individual
github.Repository{
    ...
    Owner: github.User{
        Login:"JosteinLindhom", 
        HTMLURL:"https://github.com/JosteinLindhom", 
        Type:"User"
        ...
    }
    ...
}
```

My understanding of how this works is that a repository can be owned by either an individual, or an organization. For our course repositories the owner will be an organization, whose `Login` will be equal to `course.GetOrganizationPath()`.